diff -ru a/configure.ac b/configure.ac
--- a/configure.ac	2020-05-05 15:28:06.426032419 +0000
+++ b/configure.ac	2020-05-05 15:28:52.242261793 +0000
@@ -4313,6 +4313,8 @@
 AC_SUBST(CONFIGURE, "`echo $0 | sed 's|.*/||'`")dnl
 AC_SUBST(configure_args, "`echo "${ac_configure_args}" | sed 's/\\$/$$/g'`")dnl
 
+target_cpu=`echo $target_cpu | sed s/i.86/i386/`
+
 AS_IF([test "${universal_binary-no}" = yes ], [
     arch="universal-${target_os}"
     AS_IF([test "${rb_cv_architecture_available}" = yes], [
