diff -ru a/configure.ac b/configure.ac
--- a/configure.ac	2020-05-05 15:25:33.497265676 +0000
+++ b/configure.ac	2020-05-05 15:26:41.529606987 +0000
@@ -3642,6 +3642,11 @@
 ])
 
 archlibdir='${libdir}/${arch}'
+AC_ARG_WITH(archlibdir,
+           AS_HELP_STRING([--with-archlibdir=DIR],
+                          [prefix for libruby [[LIBDIR/ARCH]]]),
+           [archlibdir="$withval"])
+
 sitearchlibdir='${libdir}/${sitearch}'
 archincludedir='${includedir}/${arch}'
 sitearchincludedir='${includedir}/${sitearch}'
